src/coreclr/src/zap/zapimage.cpp

```diff
@@ -2929,7 +2926,7 @@ HRESULT ZapImage::convertProfileDataFromV1()
                 sizeWrite += blobEntry->cBuffer;
                 if (blobEntry->blobType == ParamMethodSpec)
                 {
-                    sizeWrite -= 1;  // Adjust for
+                    sizeWrite -= 1;  // Adjust for ENCODE_METHOD_SIG prefix removal
```

Comment:
@briansull This is my best guess.  However, the code around line 3000 (search for "Adjust cSig and blob.size") removes the last byte instead of the first byte.  Is that a bug or am I missing something?

Reply:
We probably no longer have any old (circa 2012) IBC data in the V1 format, so all of this is just dead code.

Decrementing the counts here `sizeWrite `and the two counts `cSig--` and `blob.size--` both look fine to me.
It doesn't matter when you do this decrement.

Reply:
@briansull Thank you for the review.  I know we are not using the V1 format anymore; however, I think my concern is valid.  We still have C# code for V1 format, which clearly skips the first byte:

https://github.com/dotnet/runtime/blob/ac857a2e19fa42bd22ae99d12b9d9f874eba39a6/src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/IBC/IBCDataReader.cs#L597-L603 https://github.com/dotnet/runtime/blob/ac857a2e19fa42bd22ae99d12b9d9f874eba39a6/src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/IBC/IBCDataReader.cs#L443-L445

However, in zapimage.cpp we skip the last byte, because we miss `pBuffer` increment:
https://github.com/dotnet/runtime/blob/7b92525c75e245bd8badb27359ac41ef6b6ff911/src/coreclr/src/zap/zapimage.cpp#L2993-L3001

Reply:
Yes, I agree